pkg/util/cluster/cluster.go

```diff
@@ -388,22 +388,33 @@ func (c *Cluster) Delete(ctx context.Context, vnetResourceGroup, clusterName str
 		c.log.Print("deleting role assignments")
 		err = c.deleteRoleAssignments(ctx, vnetResourceGroup, *oc.OpenShiftClusterProperties.ServicePrincipalProfile.ClientID)
 		if err != nil {
-			errs = append(errs, err)
-		}
-
-		c.log.Print("deleting cluster")
-		err = c.openshiftclustersv20200430.DeleteAndWait(ctx, vnetResourceGroup, clusterName)
-		if err != nil {
+			c.log.Errorf("error when deleting role assignments: %v", err)
 			errs = append(errs, err)
 		}
 	}
 
 	if c.ci {
+		// Only perform explicit cluster deletion when in local development mode, otherwise
```

Comment:
Why doesn't deleting the resource group only work in local development mode?

Reply:
Discussed during sync call: The implementation here, with the two boolean flags (CI and localDevelopmentMode) leads to confusion on which flags are set and what behavior we expect for each permutation. We should follow-up this change with a refactor to make it more explicit and clear what steps are being performed in what contexts, even if it does lead to code duplication.